### DISCUSSION TITLE
Punctatools in 2D (single z-plane)

### DISCUSSION THREAD

**Discussion by oleland-br** [stjude/punctatools#32] [General]
The cited frontiers paper states it's possible to apply the code in 2D to a single z-plane. I need to track puncta in a time-lapse in a single z-plane. What part of the code should I modify to accomplish this?

**Comment by dwbaggett** [comment]
In theory you could probably modify punctatools to track puncta through time by analyzing each z-plane individually and then mining the output.  That said; I think you may be better served by applying the trackmate tool (https://imagej.net/plugins/trackmate/) to achieve your goal.  I've personally used trackmate to follow moving puncta in cells, and found it very good for tracking puncta of various sizes.

**Comment by oleland-br** [comment]
Trackmate would be good for the tracking aspect, for sure, but we're hoping to combine CellPost and a tracking software to track loci in cells  during the division process. Would it be easier to modify punctatools for that, or do you think developing a cellpose+trackmate workflow is more efficient?

**Reply by dwbaggett** [reply]
I see your point.  I think in that case, you could analyze the individual z-stack images using the punctatools pipeline, then during data analysis organize the data such that you could track puncta in time.  However, I think this approach may be challenging in that puncta labeling isn't guaranteed to be consistent between images, making it very hard to track specific puncta across time.  Puncta are numbered in cells based on their position, such that if a puncta is lost, or a new puncta is detected it will change the numbering of many other puncta.  

As such, before going through the trouble I would probably investigate other tools, notably there may be a Cellprofiler pipelines that could prove a good starting point: https://cellprofiler.org/examples.
